tests/shardddl1/run.sh

```diff
@@ -589,6 +589,70 @@ function DM_RENAME_COLUMN_OPTIMISTIC() {
     "clean_table" "optimistic"
 }
 
+function DM_RECOVER_LOCK_CASE() {
+    # tb1(a,b) tb2(a,b)
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(2,2);"
+
+    # tb1(a,b,c); tb2(a,b)
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add column c varchar(10);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(3,3,'aaa');"
+    check_log_contain_with_retry "putted a shard DDL.*tb1.*ALTER TABLE .* ADD COLUMN" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+
+    # tb1(a,b,c); tb2(a)
+    run_sql_source2 "alter table ${shardddl1}.${tb2} drop column b;"
+    check_log_contain_with_retry "putted a shard DDL.*tb2.*ALTER TABLE .* DROP COLUMN" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+
+    echo "restart dm-master"
+    ps aux | grep dm-master |awk '{print $2}'|xargs kill || true
+    check_port_offline $MASTER_PORT 20
+    sleep 2
+    run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml
+    check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} drop column b;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,'bbb');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(5);"
+    check_log_contain_with_retry "putted a shard DDL.*tb1.*ALTER TABLE .* DROP COLUMN" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+
+    # tb1(a,c); tb2(a,b)
+    run_sql_source2 "alter table ${shardddl1}.${tb2} add column b int;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(6,'ccc');"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(7,7);"
+    check_log_contain_with_retry "putted a shard DDL.*tb2.*ALTER TABLE .* ADD COLUMN" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+
+    # recover lock, tb1's info: (a,b,c)->(a,c); tb2's info: (a)->(a,b)
```

Comment:
Do we cover case#4 at this place?

Reply:
I think yes? Though the order cannot determine, but we don't depend on the order now